R/linear_expectation.R

```diff
@@ -86,11 +86,8 @@ LinearExpectation_calcExpectedPrice <- function(aLandLeaf, aPeriod, aScenarioInf
     yr <- get_per_to_yr(per, aScenarioInfo$mScenarioType)
     price_table <- PRICES[[aScenarioInfo$mScenarioType]]
     if(aLandLeaf$mProductName[1] %in% unique(price_table$sector)) {
-      price_table %>%
-        filter(year == yr, sector == aLandLeaf$mProductName[1]) ->
-        curr.price
-
-      all.prices$price[all.prices$year == i] <- curr.price[[c("price")]]
+      all.prices$price[all.prices$year == i] <- price_table$price[price_table$year == yr &
```

Comment:
Interesting that this is faster, since it should be doing more or less the same thing.  

One thing I had wondered, though, is whether this table changes between model runs.  If not, then maybe the thing to do is to read it once, do something like `price_tables_by_year <- split(price_table, price_table$year)`, and store that as package data.  Then the RHS of these assignments could just do something like `price_tables_by_year$year` (in this case you'd still have to filter by sector, but it looks like there are a lot of cases where we just filter by year.

Reply:
I think the speed up is from removing the filter. I don't think prices change between runs, so I think we could do the split thing. I may save that for a future PR though.

Reply:
Yeah, let's decide if we think we will be doing this often enough to justify the effort first.